lib/facter/iis_version.rb

```diff
@@ -1,24 +1,7 @@
 Facter.add("iis_version") do
  confine :kernel => :windows
   setcode do
-    iis_ver = nil
-    begin
-      require 'win32/registry'
-      access_type = Win32::Registry::KEY_READ | 0x100
-      hklm        = Win32::Registry::HKEY_LOCAL_MACHINE
-      reg_path    = 'SOFTWARE\Microsoft\InetStp'
-      reg_key     = 'VersionString'
-      
-      iis_version_text = ''
-      hklm.open(reg_path, access_type) do |reg|
-        iis_version_text = reg[reg_key]
-      end
-      if iis_version_text.match(/^Version (\d+\.\d+)$/)
-        iis_ver = $1
-      end
-    rescue
-      iis_ver = nil
-    end
-    iis_ver
+    # require 'puppet_x/puppetlabs/iis/iis_version'
```

Comment:
I think your intent here is to run the helper code to return the fact value in the future, right?

In case not, fact removal is generally considered a breaking change, at least in Puppet... so in Puppet itself we almost never remove, only add... and change format / type of return value very carefully.

Reply:
The former, not the latter.